cowprotocol/cowswap
# Limit UI prototype orders table 6 indicator status 2

# Summary

## Update 2023-03-09

Whole lot of changes

- Using spot price (non volume dependent) on table and for % difference calculations
- Added updater for fetching the spot prices, querying every 15s
- Doing a fancy calculation to get to the estimated execution price. See [here](https://www.notion.so/cownation/Execution-Price-Market-Price-90524299d1874a59bb34c658293a0316?pvs=4)
- Simplified how the % difference is calculated. Now it's based only on the estimated execution price and the spot price

Testing steps should be the same.

⚠️ Partial fills not supported!
⚠️ Still have not implemented tooltips/highlighting logic. Will be part of follow up PRs

## Original 

Continuing addressing indicator status

![Screenshot 2023-03-06 at 11 40 30](https://user-images.githubusercontent.com/43217/223101316-183c12f1-2638-4c40-8b5b-a31938c037d6.png)

![Screenshot 2023-03-06 at 11 43 24](https://user-images.githubusercontent.com/43217/223101328-fa3d3614-1051-474c-a4a1-73ad48b3b122.png)

Changes from previous:
- No longer using absolute percentage values
- When limit <= market price - use percentage as is
- When limit > market price - multiply percentage by `-1`

The formula is:

```
  if limit price > market price
    difference = ((execution price / limit price) - 1) * -1
  if limit price <= market price
    difference = (execution price / market price) - 1
```

Then we compare with the given thresholds:
- < 0.5% difference (negative percentages included): green
- 0.5 - 5% difference (boundaries included): blue
- '> 5% difference: grey

  # To Test

1. Unit tests

1. Place orders above/below market
2. Observe indicator dot color and console logs
